pkg/distributor/distributor.go

```diff
@@ -130,6 +133,21 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 			continue
 		}
 
+		entries := make([]logproto.Entry, 0, len(stream.Entries))
+		for _, entry := range stream.Entries {
+			if err := d.overrides.ValidateSample(userID, metricName, cortex_client.Sample{
```

Comment:
This method will update a counter with name starting like cortex_...., we would be nice if we could have rename the metric namespace. @gouthamve WDYT ?

Reply:
Yes, but it's kinda hard to achieve that right now specially with the usage of `promauto`.